### PR TITLE
Feat: create unit test and order

### LINKED ISSUE #26: Create Order
As a customer, I want to create an order from my cart so that I can choose the delivery address and payment method... And I want to create an order with all items from a shop.

### PR DESCRIPTION
**Pull request by Dat070601**

### CODE REVIEW

File: BookEcommerce.Test/ProductRepositoryTest.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                }
+            };
+            productRepositoryMock = new Mock<IProductRepository>();
+            productRepositoryMock.Setup(pm => pm.GetProductById(It.IsAny<Guid>())).ReturnsAsync((Guid id) => products.Single(x=> x.ProductId.Equals(id)));
+        }
+
+        [Test]
+        public async Task Get_product_when_productid_is_valid()
+        {
+            var productMockRepo = productRepositoryMock.Object;
```
Comment (levtmadden) [inline BookEcommerce.Test/ProductRepositoryTest.cs:47]: it's not a repo test for me. You should mock dbcontext, and then use repo method to get the entity. This way ^ is testing the mock, not the method logic

Comment (Dat070601) [reply BookEcommerce.Test/ProductRepositoryTest.cs:47]: I will fix it

File: BookEcommerce/Controllers/OrderController.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        }
+        [HttpPut("cancel/{orderId}")]
+        public async Task<IActionResult> CancelOrder(Guid orderId)
+        {
+            logger.LogInformation("Start To Cancel Status Order");
+            var res = await orderService.CancelOrder(orderId);
+            if (res.IsSuccess == true)
+            {
+                logger.LogInformation("Order status has been cancel!");
+                return Ok("Đã hủy đơn hàng");
```
Comment (levtmadden) [inline BookEcommerce/Controllers/OrderController.cs]: please pick up one language in your project

File: BookEcommerce.Services/CartService.cs [hunk 3]
```diff
@@ -32,103 +32,87 @@ public async Task<CartResponse> AddCart(CartRequest req, Guid cusId)
...
+                };
+            }
+            catch (Exception)
             {
                 return new CartResponse 
                 { 
                     IsSuccess = false,
-                    Message = e.Message
+                    Message = "System error, Add Product to Cart was fasle and please try again later! "
                 };
```
Comment (levtmadden) [inline BookEcommerce.Services/CartService.cs:83]: any log on this to find what actually wrong

File: BookEcommerce/Controllers/OrderController.cs (hunk 2 above)
Comment (Dat070601) [reply BookEcommerce/Controllers/OrderController.cs]: I will fix it

File: BookEcommerce.Services/CartService.cs (hunk 3 above)
Comment (Dat070601) [reply BookEcommerce.Services/CartService.cs:83]: I will LogError("System error and Exception was not found when Add Cart")

Comment (levtmadden) [reply BookEcommerce.Services/CartService.cs:83]: 2 things are required on a log: e.Message and e.StackTrace. We use it to fix bug

**Review by levtmadden — APPROVED** [review]

**Pull request merged**